File: newton_raphson.m
function newton_raphson()
    x = [input('x0: ')];
    TC = input('TC: ');
    epsilon = [];
    i = 1;
    while true
        x(i + 1) = x(i) - f(x(i)) / f_prime(x(i));
        epsilon(i) = abs((x(i + 1) - x(i)) / x(i + 1)) * 100;
        if epsilon(i) < TC
            fprintf("approximate root: %f, iterations: %d", x(i + 1), i);
            break;
        end
        i = i + 1;
    end                            
end

function result = f(x)
    result = exp(-x) - x;
end

function result = f_prime(value)
    syms x;
    y = diff(exp(-x) - x);
    result = vpa(subs(y, x, value));
end

File: newton_raphson.py
from sympy import Symbol, diff, exp

def f(x):
    result = exp(-x) - x	
    return result

def f_prime(value):
    x = Symbol('x')
    y = (exp(-x) - x).diff(x)
    result = y.subs(x, value)
    return result
	
	
def float_input(text):
    while True:
        try:
            value = float(input(text))
            return value
        except Exception as exception:
            print(exception)

def newton_raphson():
    x = [float_input("x0: ")]
    TC = float_input("TC: ")
    epsilon = []
    i = 0
    while True:
         x.append(x[i] - f(x[i]) / f_prime(x[i]))
         epsilon.append(abs((x[i + 1] - x[i]) / x[i + 1]) * 100)
         if epsilon[i] < TC:
             return {"approximate root": x[i+1], "iterations": i}
             break
         i += 1

if __name__ == "__main__":
    print(newton_raphson())
    
